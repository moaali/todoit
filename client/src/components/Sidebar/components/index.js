export { default as Categories } from './Categories'
export { default as Close } from './Close'
