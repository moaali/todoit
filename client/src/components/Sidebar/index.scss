@import 'settings/index';
@import 'tools/index';
@import 'helpers/index';

.sidebar {
  // Box Model
  width: 280px;
  height: 100vh;
  overflow: visible;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1000;
  transform: translate3d(0, 0, 0);

  // Cosmetics
  background: white;
  box-shadow: 3px 3px 2px 2px rgba($color-primary, 0.03);

  // Ease
  transition: all 180ms ease-in-out;

  @media (min-width: 992px) {
    // Box Model
    height: calc(100vh - 80px);
    top: 80px;
    z-index: auto;
  }
}

.collapsed .sidebar {
  // Box Model
  transform: translate3d(-120%, 0, 0);
}

.sidebar__wrapper {
  // Box Model
  height: 100%;
  overflow: auto;
}

.sidebar__content {
  // Box Model
  padding: 60px 0;
}

.sidebar__links-menu {
  // Box Model
  padding: 0;
  margin: 0;

  // Cosmetics
  list-style: none;
}

.sidebar__menu-item {
  // Box Model
  padding: 0 30px;
  position: relative;
  display: flex;
  flex-direction: row nowrap;
  align-items: center;
  justify-content: space-between;

  // Ease
  transition: all 180ms ease-in-out;

  &:hover {
    // Cosmetics
    background: rgba($color-primary, 0.04);
  }
}

.sidebar__link {
  // Box Model
  display: flex;
  width: 100%;
  flex-direction: column nowrap;
  align-items: center;
  text-decoration: none;
  text-transform: capitalize;
  padding: 12px 0;

  // Cosmetics
  color: $color-critical;

  .sidebar__menu-item--primary & { color: $color-primary; }
  .sidebar__menu-item--warning & { color: $color-warning; }
  .sidebar__menu-item--success & { color: $color-success; }

  &::before {
    content: '';

    // Box Model
    position: absolute;
    top: 0;
    left: -3px;
    width: 3px;
    height: 100%;
    display: block;

    // Ease
    transition: all 180ms ease-in-out;

    .sidebar__menu-item--is-active & {
      // Box Model
      left: 0;
    }

    .sidebar__menu-item--primary &,
    .sidebar__labels-menu & { background: $color-primary; }
    .sidebar__menu-item--warning & { background: $color-warning; }
    .sidebar__menu-item--success & { background: $color-success; }
  }

  .sidebar__labels-menu .sidebar__menu-item--is-active & {
    // Cosmetics
    color: $color-primary;
  }
}

.sidebar__link-icon {
  // Box Model
  margin-right: 20px;

  .sidebar__menu-item--warning & { color: $color-warning !important; }
  .sidebar__menu-item--success & { color: $color-success !important; }

  .sidebar__menu-item--primary &,
  .sidebar__labels-menu .sidebar__menu-item--is-active & {
    // Cosmetics
    color: $color-primary !important;
  }
}
