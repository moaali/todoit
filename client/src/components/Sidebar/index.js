import React, { PureComponent } from 'react'

import { Categories, Close } from './components'
import './index.scss'

class Sidebar extends PureComponent {
  render() {
    return (
      <aside className='sidebar'>
        <Close />
        <div className='sidebar__wrapper'>
          <div className='sidebar__content'>
            <Categories />
          </div>
        </div>
      </aside>
    );
  }
}

export default Sidebar;
