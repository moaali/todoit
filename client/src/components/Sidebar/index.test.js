import React from 'react';
import { Provider } from 'mobx-react';

import stores from 'stores';
import Sidebar from './';

const SidebarComponent = () => (
  <Provider store={{...stores}}>
    <Sidebar />
  </Provider>
);

describe('Sidebar', () => {
  it('should match snapshot', () => {
    const compnent = mount(<SidebarComponent />);
    expect(toJson(compnent)).toMatchSnapshot();
  })
})
