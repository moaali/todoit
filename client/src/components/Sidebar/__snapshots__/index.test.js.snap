// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot 1`] = `
<SidebarComponent>
  <Provider
    store={
      Object {
        "offcanvasStore": OffcanvasStore {},
        "searchStore": SearchStore {},
        "sidebarStore": SidebarStore {},
        "todosStore": TodosStore {
          "activeCategory": "all",
          "applyFilters": [Function],
          "currentOffset": 0,
          "displayedTodos": Array [],
          "filterByCategory": [Function],
          "filterByCompleted": [Function],
          "filterByQuery": [Function],
          "filters": Object {
            "category": "all",
            "completed": null,
            "query": "",
          },
          "findById": [Function],
          "loadChunkSize": 10,
          "loadTodos": [Function],
          "meta": Object {
            "all": 0,
            "completed": 0,
            "pending": 0,
          },
          "requestStatus": "error",
          "shownTodo": null,
          "todos": Array [],
        },
      }
    }
  >
    <Sidebar>
      <aside
        className="sidebar"
      >
        <inject-Close-with-store>
          <Close
            store={
              Object {
                "offcanvasStore": OffcanvasStore {},
                "searchStore": SearchStore {},
                "sidebarStore": SidebarStore {},
                "todosStore": TodosStore {
                  "activeCategory": "all",
                  "applyFilters": [Function],
                  "currentOffset": 0,
                  "displayedTodos": Array [],
                  "filterByCategory": [Function],
                  "filterByCompleted": [Function],
                  "filterByQuery": [Function],
                  "filters": Object {
                    "category": "all",
                    "completed": null,
                    "query": "",
                  },
                  "findById": [Function],
                  "loadChunkSize": 10,
                  "loadTodos": [Function],
                  "meta": Object {
                    "all": 0,
                    "completed": 0,
                    "pending": 0,
                  },
                  "requestStatus": "error",
                  "shownTodo": null,
                  "todos": Array [],
                },
              }
            }
          >
            <button
              className="sidebar__dismiss"
              onClick={[Function]}
            >
              <i
                className="fa fa-close fsz-lg"
              />
            </button>
          </Close>
        </inject-Close-with-store>
        <div
          className="sidebar__wrapper"
        >
          <div
            className="sidebar__content"
          >
            <inject-Categories-with-store>
              <Categories
                store={
                  Object {
                    "offcanvasStore": OffcanvasStore {},
                    "searchStore": SearchStore {},
                    "sidebarStore": SidebarStore {},
                    "todosStore": TodosStore {
                      "activeCategory": "all",
                      "applyFilters": [Function],
                      "currentOffset": 0,
                      "displayedTodos": Array [],
                      "filterByCategory": [Function],
                      "filterByCompleted": [Function],
                      "filterByQuery": [Function],
                      "filters": Object {
                        "category": "all",
                        "completed": null,
                        "query": "",
                      },
                      "findById": [Function],
                      "loadChunkSize": 10,
                      "loadTodos": [Function],
                      "meta": Object {
                        "all": 0,
                        "completed": 0,
                        "pending": 0,
                      },
                      "requestStatus": "error",
                      "shownTodo": null,
                      "todos": Array [],
                    },
                  }
                }
              >
                <ul
                  className="sidebar__links-menu"
                >
                  <inject-Link-with-store
                    category="all"
                    className="sidebar__menu-item--is-active"
                    count={0}
                    icon="database"
                    theme="primary"
                  >
                    <Link
                      category="all"
                      className="sidebar__menu-item--is-active"
                      count={0}
                      icon="database"
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      theme="primary"
                    >
                      <li
                        className="sidebar__menu-item sidebar__menu-item--primary sidebar__menu-item--is-active"
                      >
                        <a
                          className="sidebar__link"
                          href="#"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-database sidebar__link-icon"
                          />
                          <span
                            className="sidebar__link-text"
                          >
                            all
                          </span>
                        </a>
                        <span
                          className="sidebar__link-addon"
                        >
                          <Badge
                            className="badge--primary"
                            content={0}
                          >
                            <span
                              className="badge badge--primary"
                            >
                              0
                            </span>
                          </Badge>
                        </span>
                      </li>
                    </Link>
                  </inject-Link-with-store>
                  <inject-Link-with-store
                    category="pending"
                    className=""
                    count={0}
                    icon="repeat"
                    theme="warning"
                  >
                    <Link
                      category="pending"
                      className=""
                      count={0}
                      icon="repeat"
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      theme="warning"
                    >
                      <li
                        className="sidebar__menu-item sidebar__menu-item--warning "
                      >
                        <a
                          className="sidebar__link"
                          href="#"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-repeat sidebar__link-icon"
                          />
                          <span
                            className="sidebar__link-text"
                          >
                            pending
                          </span>
                        </a>
                        <span
                          className="sidebar__link-addon"
                        >
                          <Badge
                            className="badge--warning"
                            content={0}
                          >
                            <span
                              className="badge badge--warning"
                            >
                              0
                            </span>
                          </Badge>
                        </span>
                      </li>
                    </Link>
                  </inject-Link-with-store>
                  <inject-Link-with-store
                    category="completed"
                    className=""
                    count={0}
                    icon="check"
                    theme="success"
                  >
                    <Link
                      category="completed"
                      className=""
                      count={0}
                      icon="check"
                      store={
                        Object {
                          "offcanvasStore": OffcanvasStore {},
                          "searchStore": SearchStore {},
                          "sidebarStore": SidebarStore {},
                          "todosStore": TodosStore {
                            "activeCategory": "all",
                            "applyFilters": [Function],
                            "currentOffset": 0,
                            "displayedTodos": Array [],
                            "filterByCategory": [Function],
                            "filterByCompleted": [Function],
                            "filterByQuery": [Function],
                            "filters": Object {
                              "category": "all",
                              "completed": null,
                              "query": "",
                            },
                            "findById": [Function],
                            "loadChunkSize": 10,
                            "loadTodos": [Function],
                            "meta": Object {
                              "all": 0,
                              "completed": 0,
                              "pending": 0,
                            },
                            "requestStatus": "error",
                            "shownTodo": null,
                            "todos": Array [],
                          },
                        }
                      }
                      theme="success"
                    >
                      <li
                        className="sidebar__menu-item sidebar__menu-item--success "
                      >
                        <a
                          className="sidebar__link"
                          href="#"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-check sidebar__link-icon"
                          />
                          <span
                            className="sidebar__link-text"
                          >
                            completed
                          </span>
                        </a>
                        <span
                          className="sidebar__link-addon"
                        >
                          <Badge
                            className="badge--success"
                            content={0}
                          >
                            <span
                              className="badge badge--success"
                            >
                              0
                            </span>
                          </Badge>
                        </span>
                      </li>
                    </Link>
                  </inject-Link-with-store>
                </ul>
              </Categories>
            </inject-Categories-with-store>
          </div>
        </div>
      </aside>
    </Sidebar>
  </Provider>
</SidebarComponent>
`;
